packages/environment-ember-loose/__tests__/type-tests/ember-component.test.ts

```diff
@@ -69,18 +72,27 @@ import type { ComponentLike } from '@glint/template';
 
   interface YieldingComponent<T> extends YieldingComponentArgs<T> {}
   class YieldingComponent<T> extends Component<YieldingComponentSignature<T>> {
-    static template = template(function* <T>(𝚪: ResolveContext<YieldingComponent<T>>) {
-      expectTypeOf(𝚪.this).toEqualTypeOf<YieldingComponent<T>>();
-      expectTypeOf(𝚪.args).toEqualTypeOf<{ values: T[] }>();
-
-      expectTypeOf(𝚪.this.values).toEqualTypeOf<Array<T>>();
-
-      if (𝚪.args.values.length) {
-        yieldToBlock(𝚪, 'default', 𝚪.args.values[0]);
-      } else {
-        yieldToBlock(𝚪, 'else');
-      }
-    });
+    static {
+      templateForBackingValue(this, function* (𝚪) {
+        // We can't directly assert on the type of e.g. `@values` here, as we don't
+        // have a name for it in scope. However, the yields below confirm that the
+        // only thing we can legally yield to the default block is an element of the
```

Comment:
This confused me till I checked it out and messed around locally, so a small suggested tweak to the language in this comment:

```suggestion
        // have a name for it in scope: the type `T` is present on the class instance,
        // but not in a `static` block. However, the yields below confirm that the
```